### PR TITLE
python_caveats: Locate python

### PR DESCRIPTION
**Pull request by sjackman**
Fix the error:
```
brew info python
...
--HEAD
    Install HEAD version
sh: 1: python: not found
sh: 1: python: not found
==> Caveats
...
```

### CODE REVIEW

**Comment by sjackman** [comment]
```
sh: 1: python2: not found
sh: 1: python2: not found
```
🤕

File: Library/Homebrew/caveats.rb [hunk 1]
```diff
@@ -108,13 +108,23 @@ def function_completion_caveats(shell)
     end
   end
 
+  def python
+    @python ||= if OS.mac?
+      "python"
+    else
+      which("#{HOMEBREW_PREFIX}/opt/python/bin/python2") ||
+        which("python") || which("python2") || which("python2.7")
```
Comment (maxim-belkin) [inline Library/Homebrew/caveats.rb:116]: Try `which("python").to_s` and the like. `which("stuff")` returns a Pathname object

Comment (maxim-belkin) [reply Library/Homebrew/caveats.rb:116]: should this be in `extend/os/linux/caveats.rb`?

**Comment by maxim-belkin** [comment]
I've added a couple of suggestions

**Comment by sjackman** [comment]
`brew tests` failed on CircleCI, but passed on TravisCI.
```
Failures:

  1) Caveats#caveats python caveats gives commands to run when formula is keg_only
     Failure/Error: expect(caveats).to include("echo #{f.opt_prefix}/lib/python2.7/site-packages >> #{homebrew_site_packages/f.name}.pth")

       expected "This formula is keg-only, which means it was not symlinked into /tmp/homebrew-tests-20180108-18915-a... to set:\n    LDFLAGS:  -L/tmp/homebrew-tests-20180108-18915-aj0kls/prefix/opt/formula_name/lib\n\n" to include "echo /tmp/homebrew-tests-20180108-18915-aj0kls/prefix/opt/formula_name/lib/python2.7/site-packages >> /tmp/homebrew-tests-20180108-18915-aj0kls/prefix/lib/python2.7/site-packages/formula_name.pth"
       Diff:
       @@ -1,2 +1,6 @@
       -echo /tmp/homebrew-tests-20180108-18915-aj0kls/prefix/opt/formula_name/lib/python2.7/site-packages >> /tmp/homebrew-tests-20180108-18915-aj0kls/prefix/lib/python2.7/site-packages/formula_name.pth
       +This formula is keg-only, which means it was not symlinked into /tmp/homebrew-tests-20180108-18915-aj0kls/prefix,
       +because some reason.
       +
       +For compilers to find this software you may need to set:
       +    LDFLAGS:  -L/tmp/homebrew-tests-20180108-18915-aj0kls/prefix/opt/formula_name/lib
     # ./test/caveats_spec.rb:241:in `block (4 levels) in <top (required)>'
     # ./test/spec_helper.rb:101:in `block (2 levels) in <top (required)>'
     # ./vendor/bundle/ruby/2.3.0/gems/rspec-wait-0.0.9/lib/rspec/wait.rb:46:in `block (2 levels) in <top (required)>'

  2) Caveats#caveats python caveats when f is not keg_only gives commands to run when python pth files are installed
     Failure/Error: expect(caveats).to include(".pth files to Homebrew's site-packages and your\nPython isn't configured")

       expected "" to include ".pth files to Homebrew's site-packages and your\nPython isn't configured"
       Diff:
       @@ -1,2 +1 @@
       -.pth files to Homebrew's site-packages and your\nPython isn't configured
     # ./test/caveats_spec.rb:228:in `block (5 levels) in <top (required)>'
     # ./test/spec_helper.rb:101:in `block (2 levels) in <top (required)>'
     # ./vendor/bundle/ruby/2.3.0/gems/rspec-wait-0.0.9/lib/rspec/wait.rb:46:in `block (2 levels) in <top (required)>'

  3) Caveats#caveats python caveats when f is not keg_only give commands to run when Homebrew's site-packages is not in Python sys.path
     Failure/Error: expect(caveats).to include("Homebrew's site-packages is not\nin your Python sys.path")

       expected "" to include "Homebrew's site-packages is not\nin your Python sys.path"
       Diff:
       @@ -1,2 +1 @@
       -Homebrew's site-packages is not\nin your Python sys.path
     # ./test/caveats_spec.rb:220:in `block (5 levels) in <top (required)>'
     # ./test/spec_helper.rb:101:in `block (2 levels) in <top (required)>'
     # ./vendor/bundle/ruby/2.3.0/gems/rspec-wait-0.0.9/lib/rspec/wait.rb:46:in `block (2 levels) in <top (required)>'

Finished in 1 minute 41.8 seconds (files took 1 minute 4.11 seconds to load)
30 examples, 3 failures, 2 pending

Failed examples:

rspec ./test/caveats_spec.rb:234 # Caveats#caveats python caveats gives commands to run when formula is keg_only
rspec ./test/caveats_spec.rb:225 # Caveats#caveats python caveats when f is not keg_only gives commands to run when python pth files are installed
rspec ./test/caveats_spec.rb:219 # Caveats#caveats python caveats when f is not keg_only give commands to run when Homebrew's site-packages is not in Python sys.path
```
https://circleci.com/gh/Linuxbrew/brew/745

**Comment by sjackman** [comment]
I've merged the related PR https://github.com/Linuxbrew/brew/pull/560

**Comment by sjackman** [comment]
I'm closing this PR unmerged in favour of PR #560.

**Pull request closed without merging**